tests/types.rs

```diff
@@ -91,3 +91,19 @@ fn test_using_cql_protocol_version() {
     cluster.set_protocol_version(4).unwrap();
     cluster.set_protocol_version(2).unwrap();
 }
+
+#[test]
+fn test_parsing_printing_batch_type() {
```

Comment:
:-) `variants()` is created by the macro I wrote in `utils.rs` - it's not a standard Rust thing (it can't be, because you couldn't do that for enum variants with arguments).